app/presenters/document_presenter.rb

```diff
@@ -8,13 +8,31 @@ class DocumentPresenter
     :bulk_published,
     to: :"document.details"
 
+  attr_reader :finder
+
   def initialize(finder, document)
     @finder = finder
     @document = document
   end
 
+  def respond_to?(method)
+   if finder.facets.map(&:key).include?(method.to_s)
```

Comment:
Is it worth extracting this condition and the one on line 27 into a method so you can call `if facet_keys.include?(method.to_s)` or `if has_facet?(method.to_s)`